### DISCUSSION TITLE
Confusing warning message "deprecated convert command" even using "magick convert" and correct usage for sub-commands in IMv7

### DISCUSSION THREAD

**Discussion by uprprc777** [ImageMagick/ImageMagick#7366] [Help] [answered]
### ImageMagick version

7.1.1-33 Q16-HDRI aarch64 22263 (from Homebrew)

### Operating system, version and so on

macOS 14.5 (23F79)

### Description

```
$ magick convert
WARNING: The convert command is deprecated in IMv7, use "magick"
```

```
$ convert
WARNING: The convert command is deprecated in IMv7, use "magick"
```

I am already using "magick" as the main command while it still triggers the warning message.
Is it a problem with ImageMagick or my usage?

Please help!
Thank you!

**Comment by dlemstra** [comment]
You should not use `magick convert` but just `magick`. We can update the message to make this more clear.

**Reply by iatomic1** [reply]
I just installed betterlockscreen how can i change it to just magick?

**Reply by uprprc777** [reply]
Thank you for your help!

However, I am still a bit confused.

Specifically, if my original command for ImageMagick 6 is:

```
magick convert -quiet -regard-warnings input_file -define png:compression-level=0 -define png:compression-filter=0 output_file.png
```

For IM 7, is the following usage correct?

```
magick -quiet -regard-warnings input_file -define png:compression-level=0 -define png:compression-filter=0 output_file.png
```

And for other sub commands like `identify`, is `magick SUB_COMMAND ...`, `magick identify ...` the correct and recommended usage?

I found something like `-identify` in `magick` (IM7) and read the [official manual](https://imagemagick.org/script/command-line-options.php#identify).

I tried some command myself but all failed.

Something like the following failed with IM7

```
magick -identify -quiet -format %# /path/to/an/image.png
```

While the following is OK with IM6 (also IM7):

```
magick identify -quiet -format %# /path/to/an/image.png
```

**Comment by fmw42** [comment]
magick convert is not fully IM 6, nor IM 7.  For IM 6 use

`convert -quiet -regard-warnings input_file -define png:compression-level=0 -define png:compression-filter=0 output_file.png`

Try identify not -identify

`magick identify -quiet -format %# /path/to/an/image.png`

**Reply by uprprc777** [reply]
Thank you for your reply!

Is this "fully" IM 7?

```
magick -quiet -regard-warnings input_file -define png:compression-level=0 -define png:compression-filter=0 output_file.png
```

And for the identify,

```
magick identify -quiet -format %# /path/to/an/image.png
```

Before I made this post I used this a lot and it worked well without any error or warning.
But I want to know if `magick identify ...` is recommended usage in IM 7?

In the official manual for IM7, there is a section for `-identify` for `magick` but I don't understand.
And in [here](https://imagemagick.org/script/command-line-tools.php), it says `magick identify` is "support sub-commands for compatibility with ImageMagick [version 6](http://legacy.imagemagick.org/)".

I am confused if `magick identify` is deprecated in IM7 and what's the best usage?

**Comment by snibgo** [comment]
> magick convert -quiet -regard-warnings input_file -define png:compression-level=0 -define png:compression-filter=0', output_file.png

That command contains a superfluous quote and comma `',`. Remove them.

**Reply by uprprc777** [reply]
thank you for noticing this typo!

**Comment by fmw42** [comment, marked as answer]
Yes,  use only magick in IM 7 when you want to do the equivalent of convert in IM 6.  Preface all other IM 6 tools with magick in IM 7.  So use magick identify, magick mogrify, magick montage, etc.

**Reply by hackerb9** [reply]
Thank you, Fred. It is frustrating that the interface is inconsistent. I had hoped I could have both IM6 and IM7 compatibility by simply doing:

```bash
magick=$(type -p magick)
$magick identify ...
$magick convert ...
$magick montage ...
```

but now I have to figure out something more complex. Your suggestion of `ln -s magick convert` is a good one for now, but won't work once ImageMagick deprecates the other legacy commands.

Do the ImageMagick developers have documentation explaining these changes?  In particular, I want to know what they suggest folks who have built up programs around ImageMagick for decades should do. I understand that simply not upgrading to IM7 is a fine option for many people — Debian only ships with IM6, for example — but I'd like my scripts to work no matter which version of ImageMagick is installed.

**Reply by fmw42** [reply]
For my scripts, I only use convert. But I rely upon the copy that is made in the utilities directory to make convert be the same as magick.  This only happens for versions of IM for which I do that, unfortunately, and I have to make the copy each new release of IM 7.

For documentation on IM 7 changes, see https://imagemagick.org/script/porting.php#cli.  Basically only convert was upgrade in IM 7 as magick. But all the other tools are still basically IM 6. In order to use them under IM 7, one prefaces each with magick, except convert, which was improved.  All the other tools have not been changed from IM 6.  This way you do not have to have IM 6 and IM 7 on your system to run the other tools. However, in your case, you probably need both IM 6 and IM 7.  That is what I do so I can test my scripts under both versions.

There is a transportable version of IM 7 binary for Linux, called magick, that one can download and put anywhere you want that does not conflict with IM 6.  See https://imagemagick.org/script/download.php

**Reply by uprprc777** [reply]
Thank you, Fred!

Your reply has clarified everything for me. I marked your reply as the answer.

By the way, I think moving `convert` to `magick convert` instead of `magick` would be better, as it is more consistent with other tools like `magick identify`, `magick montage`, etc., making (almost) all tools called in `magick SUB_COMMAND` form.
Though, it's good to have `magick identify`, `magick montage`, etc. replacing `identify`, `montage`, etc., and `magick` is at least better than `convert`.

**Reply by fmw42** [reply]
We need both for those who want to call convert from IM 6 on a system that has IM 7 as well.

**Reply by hackerb9** [reply]
@uprprc777 wrote:
> By the way, I think moving `convert` to `magick convert` instead of `magick` would be better, as it is more consistent

Agreed. I think a good idea would be to simply have  `magick convert` work the same as `magick` without any deprecation message, but that's up to @dlemstra and the rest of the ImageMagick dev team.

@fmw42 Thank you for the ImageMagick 7 documentation link. It is interesting reading, but unfortunately does not seem to cover what to do to for scripts that could run fine in either IM7 or IM6, but currently break due to the name changes. 

Here's the solution I came up with which can be put at the top of any Bash script:

```bash
shopt -s expand_aliases         # Allow aliases for working around quirks.

# For ImageMagick 6 <-> 7 compatibility.
if type magick &>/dev/null; then
    alias convert='magick'
    alias montage='magick montage'
fi
```

The rest of the script can remain as it is, happily continuing to use `convert`.

**Reply by FrontierDK** [reply]
I just discovered this as well, using the DLL version. I don't mind that methods are added, but for the love of God, why did the old methods have to be ruined?

**Reply by hackerb9** [reply]
I don't know. It seems like it will only discourage people from installing the latest version of ImageMagick since (from my perspective) it is needlessly breaking the previous API.

I say "needlessly" knowing that there probably was some need I am unaware of. If not needless, though, it is certainly perplexing given the simple ~~fixes~~ kludges both I and @fmw42 have offered for backwards compatibility.

Whatever the reason, I can't find it, and maybe that is a hint at a more important issue. The new documentation has a poor signal-to-noise ratio -- the needle I seek may or may not be in there, but only the few who have made it through the whole haystack know for sure. It makes me wonder if not enough people are contributing to the project as documentation authors and editors.

**Reply by snibgo** [reply]
> ... it is needlessly breaking the previous API.

The C-level API changed from v6 to v7, but there have been very few queries in these forums about that.

The Command Level Interface (CLI) changed from v6 to v7 partly because the old v6 operation sequence was increasingly difficult to maintain, document, explain, or justify. Now the operation sequence is really simple: operations are done in the same order they appear on the command line. Yes, this broke scripts that (for example) blurred an image before reading the image. There were also other changes, such as allowing %[fx:...] in most places that previously required decimal numbers.

When the developers did this, they said they would support the old syntax for ten years. This would be plenty of time for script-writers to adopt the new syntax. (At the time, I thought, surely no-one needed such a long overlap? I was wrong.)

**Reply by fmw42** [reply]
I think the main reason that people still need V6 is that the maintainers of Linux don't seem to want to upgrade to V7 for reasons beyond me.

**Reply by FrontierDK** [reply]
I use the .DLL version on Windows, and even there, I get the message in the console. -quiet does nothing.

**Reply by hackerb9** [reply]
@snibgo: Changing the operation sequence broke scripts and there was a clearly articulated need for why. That's a good contrast with the issue at hand: The convert command is issuing deprecation messages which has broken scripts and it is not clear why doing so was necessary. 

@fmw42 I can't speak for the Linux distributions, but personally I tried IM v7 years ago and had to revert because it was not designed to coexist with v6. I seem to recall that you, Dr. Weinhaus,  found ways around that by using unique pathnames, but did your ideas ever got folded back into ImageMagick? That's the kind of thing distributions often care about.

**Reply by fmw42** [reply]
No actually I have a script that allows me to run IM 7 even with IM 6. I have to install from source and then configure and then only make, no sudo make install.

**Reply by Scal-Human** [reply]
This is marked as the answer while it does not answer anything.
Simply removing convert DOES NOT work.
I was doing
```
magick.exe convert -background white -transparent white -resize 256 user.svg -fill LimeGreen -opaque black -define icon:auto-resize=96,32 -colors 256 user.ico
```
Nope, cannot get it to work.
Does somebody know of a tool to convert svg to ico under Windows ?

**Reply by fmw42** [reply]
**On IM 7, use magick, not magick convert**.  Magick convert is like convert, which has been deprecated as it is now very old and 10 years past IM 7 introduction.

**Reply by snibgo** [reply]
@Scal-Human: What error message(s) do you get? Does it make a wrong output image file, or no output image file? Without such information, we are forced to guess.

"magick" without the "convert" will use IM V7 syntax. This will execute operations in the same order that you supply them in the command. But you have "-resize" before you have read an image. I expect you get an error message like "no images found for operation `-resize'". You need to read the image, then resize it.

**Reply by Scal-Human** [reply]
I have the warning, no output file, no other error message.
Do you have another tool to propose ?
It takes a while to obtain the correct combination of parameters and now I have to loose time I already spent for the same result.
After this and the previous [issue about FFmpeg](https://github.com/ImageMagick/ImageMagick/issues/6875) on which I have lost time also and that was due to the aggressive insertion at the beginning of the PATH, I do not want to continue to use magick anymore.

**Reply by hackerb9** [reply]
@Scal-Human wrote:
> I do not want to continue to use magick anymore.

I can understand not wanting ImageMagick 7 given the compatibility issues. I'm not saying you should stick with ImageMagick if you don't want to, but may I suggest giving the "Legacy" version of ImageMagick 6 a try? I'm one of many people who happily use it. You can get it from here: https://legacy.imagemagick.org/script/download.php .

**Reply by Scal-Human** [reply]
@hackerb9 Thank you for the link, but 1) is it not a version that has an ancient FFmpeg 4.x in it ? 2) it breaks the update process (update all ... except this one).

**Reply by hackerb9** [reply]
Those are good questions which unfortunately I cannot answer as I use the packaged version included with Debian GNU/Linux. I had no idea ImageMagick ever included ffmpeg. I also do not know what you mean by "update process". Does Microsoft Windows come with a package manager now? 
_____
[EDIT]

I see in you other bug report that you mention Chocolatey. Did you try `choco remove imagemagick` so the package manager won't try to update ImageMagick? Alternately, what about something like `choco install imagemagick --version=6.9.12.7`?

**Reply by dlemstra** [reply]
Moving back to ImageMagick 6 is probably not your best future proof decision. We will not add new features/formats to ImageMagick 6 anymore. We now only fix bugs and security issues. But that is also subject to change in the future if we decide to only do security fixes or completely stop working on that version. You now had 9 years to move to this version so I would recommend you to make this move now.

**Reply by hackerb9** [reply]
> Moving back to ImageMagick 6 is probably not your best future proof decision. 

😆 Hah, probably not!

My comment was directed to @Scal-Human who is so frustrated with IM 7 that they don't want to use ImageMagick _at all._ I was suggesting IM 6 as a way they can get their work done. I don't suggest it for new scripts.


> You now had 9 years to move to this version so I would recommend you to make this move now.

Okay, about that... 

IM 7 is great and the work you've done is impressive. I don't want anyone to think I am complaining, yet I have to correct this as I'm surprised that you, @dlemstra , of all people, would think that people have had time for a transition.

I've been waiting for IM 7 to be ready for nearly a decade and, at least for me, it still isn't. There is no standard way of having both IM 6 and 7 installed.

Think about it this way: If a new version of a library is installed, even if the API changes, it doesn't break your existing programs. Those programs can still be linked to the old version. Sites can upgrade their programs one at a time, when it is convenient for them, and use the new library when writing new programs. Easy-peasy.

ImageMagick doesn't work that way.

Command line programs don't have library versioning, so their interfaces rarely change in breaking ways, even when obvious improvements could be made (take `dd`). When something comes out that is improved to the point that it breaks compatibility it often gets a new name (`more` and `less`).

Consider the Python 2 to Python 3 transition. As you may recall, the solution was to have both installed simultaneously and let scripts specify which one to use (`#!/usr/bin/python3`). It still took a decade, but the clock didn't start on the transition until after a solution was in place.

Please do not think that people have had "9 years". There is still no path for a smooth transition to IM 7 and slow phasing out of IM 6.

I appreciate that it looks like IM 7 is moving towards what I'm talking about. Once it no longer provides `convert` and friends, hopefully IM 6 and 7 will be able to coexist. And then we can start the clock on how long people have had to transition. 🙂

Thanks again for all the work you do on ImageMagick.

**Reply by snibgo** [reply]
> There is no standard way of having both IM 6 and 7 installed.

I don't know about a *standard* way. On Windows 11, I have multiple IM installations, including some v6 and some v7. For some years, I wrote software (in C or C++) such that it could be compiled either with v6 or v7. A year or three ago, I decided there was no longer a need for this, and I now write for v7 only.

From 2016 to 2018, I edited my [IM web pages](http://im.snibgo.com) to use v7 `magick` instead of v6 `convert`. The elapsed time was 2 years, but only a couple of weeks of actual work.

I have never used "`magick convert`", where the `magick` program processes its arguments under the old v6 syntax. It is a workaround that I have not needed.

So, for me, the transition from v6 to v7 was quite simple.

**Reply by Scal-Human** [reply]
@hackerb9 Indeed I use Chocolatey, and when I am talking about update process, it is a series of scripts that update a lot of different things from different sources on multiple machines, but each source is updated as a whole, e.g. Chocolatey: **choco upgrade all**.
I just check before updates to see what have change, read the release notes, then I launch all updates.
Having to "not update" or "update to a specific version" is easy to say, but how I can upgrade all except magick ... it breaks the update process that must be reviewed, rewritten, retested ... possibly splitted application by application ... it completly defeats the purpose of package/application manager.
Of course, one could also say: "you can update everything manually, dude" (I write it myself to avoid another dude to answer that).
And, yes, @hackerb9 , it is frustrating ... (and I deleted the rest of the sentence).
To be complete: I am not an image processing specialist, so arriving at the result I want takes time, learning, reading, testing.

Because of you, @hackerb9, and your eager attempts to help me (thanks, by the way), I gave it another shot.
I write it below in case it can help somebody else.
So, my command was
```
magick.exe convert -background white -transparent white -resize 256 user.svg -fill LimeGreen -opaque black -define icon:auto-resize=96,32 -colors 256 user.ico
```
I honestly do not remember why the parameters were in that order (I did it a while ago).
Removing the **convert** as suggested does not work.
I tried this below (only change other than removing convert is the position of the input svg)
```
magick.exe -background white user.svg -transparent white -resize 256 -fill LimeGreen -opaque black -define icon:auto-resize=96,32 -colors 256 user.ico
```
This works, and I do not want to know why.
Thanks @hackerb9

**Comment by MrJake222** [comment]
Hi,

This is confusing. `magick -auto-orient *.jpg out.pdf` fails with `magick: no images found for operation `-auto-orient' at CLI arg 1 @ error/operation.c/CLIOption/5479.`

This works though: `magick convert -auto-orient *.jpg out.pdf`, but I get this annoying warning. Using: `Version: ImageMagick 7.1.1-33` (Arch).

**Reply by fmw42** [reply]
In IM 7 use magick, not magick convert.  Read the input image right after magick and then put the -auto-orient after the image.  Imagick 7 is strict on the order of things in the command line. (For raster files, it should be: input, settings, operators, output)

**Reply by pefu** [reply]
This is an incompatible command line syntax change with earlier versions of imagemagick. So simply replacing "convert" with "magick" breaks programs. For example the tool easygg which I used for decades to create thumbnails and slideshow images for my fotos stopped working after I changed "convert" into "magick". I had to go back to "magick convert" to make it working again.

**Comment by installgentoo** [comment]
Changing convert to magick in my scripts makes the scripts fail with

```magick: no images found for operation `-alpha' at CLI arg 3 @ error/operation.c/CLIOption/5481.```

Why do you need to break the tool for this busywork rubbish?

**Comment by snibgo** [comment]
In v7 `magick` (which has now been available for ten years), options are executed in the order they appear in the command. For example: read an image, process it, write it to a file. We can't process an image (eg with `-alpha`) before it has been read.

The old v6 `convert` syntax processed options in some order that was difficult for humans to predict.

There have been many other improvements in v7. When upgrading from v6 to v7, I suggest reading https://www.imagemagick.org/script/porting.php

**Comment by rubyFeedback** [comment]
> There have been many other improvements in v7. When upgrading from v6 to v7, I suggest reading https://www.imagemagick.org/script/porting.php

I think you need to understand the user-side and developer-side too. The first time I noticed something changed was due to the annoying deprecation warnings. So, only via Google I then found the discussion here - which pointed to the porting.php page. Is everyone else able to find that? 

I also had a looking at porting.php. Unless I have missed it that page does not even mention "convert" being obsoleted at the time of me writing this comment. So I am not sure if that porting.php is too terribly useful in regards to this issue here. Perhaps you guys have not considered the possible impact deprecations have on downstream code? Otherwise why isn't it like the number one point to talk about on the homepage? 

Anyway, I am willing to adapt and adjust my code, so I finally decide to rewrite my large ruby-based wrapper project over imagemagick. That also helps polish old things and add new things too, even though it costs a bit of time. My request, though, is that you guys can be very careful about what to deprecate, when and how. My terminal output was filled with tons of deprecation warnings since I used to call "convert" directly. I found the issue request here by google searching via "imagemagick convert is deprecated".

**Reply by fmw42** [reply]
Look at https://imagemagick.org/script/porting.php#deprecate "Command Changes".  The first line is about Magick replacing convert.  It is in bold.

**Reply by skin-soc** [reply]
What a totally unnecessary mess

**Reply by DcsMarionDickten** [reply]
My apologies for butting in... RubyFeedback requested putting these changes prominently directly onto the homepage, not onto a subpage only found via googling the problem that arose because they had never seen there were to be breaking changes. (Just mentioning it because I'm in exactly the same situation, sorry again. I agree with all your points. It's just such a jolting experience for a developer. Being under constant time pressure, I really appreciate advance warnings in places I visit anyway, e.g. the homepage.)

**Comment by suugbut** [comment]
Just for documentation:

version 6: `magick convert -density 100 -alpha remove input.pdf output%02d.png`

must be changed as follows **(the input/output position must be rearranged as well)**.

version 7: `magick -density 100 input.pdf -alpha remove  output%02d.png`


Without the rearrangement, it triggers the **BELOVED** warning: `magick: no images found for operation `-alpha' at CLI arg 3 @ error/operation.c/CLIOption/5481.`

**Comment by J-Dunn** [comment]
Imagemagick is an impressively powerful tool rendered nearly useless by a total lack of overall command structure design. 
It seems that everything just kind of evolved by a series of bolt-on tweaks, hacks and work-arounds and requires a mental stack depth of a chess grand master to use it. 

> The old v6 convert syntax processed options in some order that was difficult for humans to predict.

... or remember. 

It is arcane esoteric and a PITA to use.  It is as user friendly as programming directly in machine code without the help of an assembler, except that machine language usually has some well defined structure behind it. ( I'm not joking, I learnt to progam Z80 writing hex files by hand). 

Hopefully IM7 is creating something of the missing unified command structure. 

> My comment was directed to @Scal-Human who is so frustrated with IM 7 that they don't want to use ImageMagick at all. 

I arrived at that stage with IM6. It is an incredibly powerful stack of tools whose usefulness is severely limited by lack of top down design. 

I'm sure the reordering will help but what was probably needed was a total rewrite, which would probably not have required more downstream effort than the way this was rolled out. 

/m2c

**Comment by urban-warrior** [comment]
IM6 evolved through bolt‑ons, which made the CLI unpredictable. ImageMagick 7 (IM7) is a complete rewrite with a unified design. It treats the command line as a scripting language, so instead of juggling arcane option order, you can write clear, structured sequences of operations. This makes IM7 far more coherent and practical for complex workflows than IM6 ever was. Some may feel the parser could be improved, and we agree. Since IM7 is open source, you’re welcome to develop your vision of a parser and submit it as a pull request for review and possible inclusion in a future release.

**Reply by J-Dunn** [reply]
thanks for the reply. Maybe this is what it needed all along and may address many of my gripes.

**Comment by J-Dunn** [comment]
Since these syntax changes are likely to throw up lots of issues some more helpful error context may be useful. 

I tried a third party script which I'd used quite a lot in the past.

I changed convert to magick and _identify_ to _magick identify_  and tried to test it again and it threw me this:
`magick: Not a real operator at ')' @ error/fx.c/GetOperator/2568.`

The line number is internal to IM7 not the input it is running . 
The only clue I get to go and search for the problem line in ONE character: ')'
I'm guessing there are rather a lot of occurrences of that one char. 

That is not really that helpful.

**Comment by snibgo** [comment]
The error message has come from fx.c. Hence the problem is from either `-fx` or `%[fx:...]`. The close parenthesis occurs at the end of the fx expression.

V6 had poor error checking, and permitted fx expressions that were not valid. Since then, fx error checking has been improved. So you should examine your script for invalid fx expressions.

EDIT: If you post the script, or just the IM command that has the problem, I can look at it to suggest where the error is, and perhaps to improve the fx error message.

**Comment by J-Dunn** [comment]
Thanks, It was Fred's cameradeblur which was producing that error.

https://www.fmwconcepts.com/imagemagick/cameradeblur/index.php

**Comment by fmw42** [comment]
I tested both camerablur and cameradeblur yesterday with both IM 6 and IM 7 and for me it was camerablur that had the ")" error and was in IM 7. IM 6 worked for both scripts.  Cameradeblur had some other error that turns out was in Imagemagick 7 that has now been fixed and is supposed to be out in the potential release today. Perhaps when I can test it again with the new release, I might find the ')' but in cameradeblur.  

But be sure you are using the latest versions.

**Comment by J-Dunn** [comment]
Hi Fred, I'm using distro supplied  ImageMagick 7.1.1-47 Q16-HDRI x86_64 22763


```
 ./cameradeblur -a 12  -n 0.015 bather.png bather_a12_n015.png 
magick: Not a real operator at ')' @ error/fx.c/GetOperator/2568.
```

**Reply by fmw42** [reply]
Post your bather.png so that I can test with it.  But how recently did you download my cameradeblur script?  Also 7.1.1.47 is a bit old.  I tested with IM 7.1.2.27. The version of cameradeblur that I tested was revision date of 5/15/2023. I found no extra ')' in it.

**Reply by J-Dunn** [reply]
I think I may have induced the problem.  I downloaded the script again yesterday and IM7 was spamming the console with deprecation warnings. 
```


WARNING: The convert command is deprecated in IMv7, use "magick" instead of "convert" or "magick convert"

WARNING: The convert command is deprecated in IMv7, use "magick" instead of "convert" or "magick convert"

WARNING: The convert command is deprecated in IMv7, use "magick" instead of "convert" or "magick convert"

WARNING: The convert command is deprecated in IMv7, use "magick" instead of "convert" or "magick convert"

WARNING: The convert command is deprecated in IMv7, use "magick" instead of "convert" or "magick convert"

WARNING: The convert command is deprecated in IMv7, use "magick" instead of "convert" or "magick convert"

WARNING: The convert command is deprecated in IMv7, use "magick" instead of "convert" or "magick convert"

```
I do get an output file. 

I started editing to follow that recommendation.  It does not warn me that the syntax and argument order has changed ! 

I just pulled it again and I don't get the error, just a screen full of WARNING:

Sorry for inaccurate info but since you found a bug in camerablur and there was a bug in IM7 as well, it was not a waste of effort. 

Can I tidy this up to work on IM7 if I replace  "convert" with "magick convert" ? 
It is unclear what that warning is suggesting. 

If I follow the changes, what it needs is a rewrite to use magick but with new syntax or to replace  "convert" with "magick convert" which conserves the old syntax. 

Is that correct ?

**Reply by fmw42** [reply]
@J-Dunn. You could replace convert with magick, **not** magick convert. But there may be identify in the code and that would have to be replace with magick identify. You might also try making a symbolic link from convert to magick, if you do not need to use IM 6 any longer.

**Reply by fmw42** [reply]
@J-Dunn  Please post your bather.png so that I can test with it

**Reply by J-Dunn** [reply]
> @J-Dunn. You could replace convert with magick, **not** magick convert. But there may be identify in the code and that would have to be replace with magick identify. You might also try making a symbolic link from convert to magick, if you do not need to use IM 6 any longer.

Thanks. My distro already does that simlink. 

So the message means :  (use "magick" instead of "convert" ) or use "magick convert" with new syntax . 
Is that right?

**Reply by fmw42** [reply]
Use magick in place of convert, NOT magick convert.

**Comment by J-Dunn** [comment]
This fixes the warning spam.  I did not change identity, (3x IIRC). 
`sed -i s/convert/magick/g cameradeblur`

Seems to still function as before. 
Thanks for the help. It would be a shame not to have these great scripts. 

Some thought is needed to make the warning clearer if it is supposed provide a remedy without a full update.

**Reply by fmw42** [reply]
The warning comes from Imagemagick.  It is because my script uses convert because it was writing before IM 7. I would think as symbolic link from convert to magick would have solved the issue. Are you sure, you already had that symbolic link?

**Reply by fmw42** [reply]
Yours is the second report of the ")" issue in cameradeblur.  So either you were using an older version of my script or there is still something wrong in the script. If it actually reported that issue, then I need your image for which it failed and the command line that you used.

> ./cameradeblur -a 12  -n 0.015 bather.png bather_a12_n015.png 
> magick: Not a real operator at ')' @ error/fx.c/GetOperator/2568.

**Reply by J-Dunn** [reply]
> Are you sure, you already had that symbolic link?

Yes, I did not create recently . It seems to be a distro tweak. 

```
ls -l `which convert`
lrwxrwxrwx. 1 root root 6 Mar 30  2025 /bin/convert -> magick
```
Despite the simlink , it still seems to detect that the calling command line was convert: 
```
$ convert > /dev/null
WARNING: The convert command is deprecated in IMv7, use "magick" instead of "convert" or "magick convert"

convert: missing required argument  @ error/deprecate.c/ConvertImageCommand/571.
```

Don't worry about the ')' error , as I posted above, I think it was due to my poking around to remove the warning spam and misunderstanding  the wording of the warning.

**Reply by fmw42** [reply]
Another solution for the convert deprecation warning would be to edit my script and just put the following right after the #!/bin/bash.

```
convert() {
    # -quiet tells ImageMagick to suppress warning logs natively
    magick -quiet "$@"
}
export -f convert

```

This code will change convert to "magick -quiet "$@". -quiet suppresses warninsg and $@ intercepts convert and changes it to the magick -quiet.

Or if you do not want to suppress all warnings, then you could do the same but with the following code:

```
convert() {
    # Runs magick, but silences the specific deprecation warning from stderr
    magick "$@" 2> >(grep -v "WARNING: The convert command is deprecated in IMv7, use "magick" instead of "convert" or "magick convert" >&2)

```

<br>

Be sure that the warning in the above command is exactly what you get.

If you try either, let me know if it works?

**Comment by J-Dunn** [comment]
`sed -i s/convert/magick/g cameradeblur`
Thanks for the alternative but sed fixed it.

identify does not seem to produce warnings, so I did not alter that. 

Thanks for your rapid and enthusiastic attention to this problem.

**Reply by fmw42** [reply]
I already had fixed identify with a switch conditional so that it would handle IM 6 convert or IM 7 magick.
